pkgs/applications/editors/eclipse/plugins.nix

```diff
@@ -1,9 +1,28 @@
-{ lib, stdenv, fetchurl, fetchzip, unzip }:
+{ lib, stdenv, fetchurl, fetchzip, zip, unzip, xmlstarlet, eclipses }:
 
 rec {
 
-  # A primitive builder of Eclipse plugins. This function is intended
-  # to be used when building more advanced builders.
+  sanitizeUpdateSite = outDir: ''
+    workDir=/build/sanitizeTmp
+    mkdir $workDir
+    pushd $workDir
+    unzip "${outDir}/artifacts.jar"
+    unzip "${outDir}/content.jar"
+    xmlstarlet edit \
+      --inplace \
+      --update "/repository/properties/property[@name='p2.timestamp']/@value" \
+      --value "0" \
+      artifacts.xml content.xml
+    touch artifacts.xml content.xml -t '197001010100'
+    zip -X - artifacts.xml > "${outDir}/artifacts.jar"
+    zip -X - content.xml > "${outDir}/content.jar"
+    popd
+  '';
```

Comment:
This feels a bit ad hoc.

Both these jars just contain a single xml file, and the xml files contain an attribute with a timestamp, so we:
- extract
- zero the xml timestamp with xmlstarlet
- clear file timestamps
- rezip

Is there a cleaner way to do any of this? Existing utilities in nixpkgs to help?